camelConnector/src/test/java/io/vantiq/extsrc/camelconn/connector/HandleConfigurationWithOverrides.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 Vantiq, Inc.
+ *
+ * All rights reserved.
+ *
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.camelconn.connector;
+
+import io.vantiq.extsrc.camelconn.discover.CamelRunner;
+import io.vantiq.extsrc.camelconn.discover.CamelRunnerWithComponentSubs;
+import org.apache.camel.Component;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Test version of CamelHandleConfiguration.
+ * </p>
+ * This provides a type-safe, test only, mechanism to allow component type overrides for testing.  We do this to
+ * handle, amongst other things, Vantiq-server-less implementation of the Vantiq Camel Component for use in unit tests.
+ */
+public class HandleConfigurationWithOverrides extends CamelHandleConfiguration {
+    
+    Map<String, Component> overrides = null;
+    
+    public HandleConfigurationWithOverrides(CamelCore source) {
+        super(source);
+    }
+    
+    public HandleConfigurationWithOverrides(CamelCore source, Map<String, Component> overrides) {
+        super(source);
+        this.overrides = overrides;
+    }
+    
```

Comment:
override creating of CamelRunner to get one with component overrides.  The components overridden provide unit test versions that don't require an operating Vantiq server.